plugins/modules/host_errata_info.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2021 Evgeni Golov
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: host_errata_info
+version_added: 2.1.0
+short_description: Fetch information about Host Errata
+description:
+  - Fetch information about Host Errata
+author:
+  - "Evgeni Golov (@evgeni)"
+options:
+  host:
+    description:
+      - Name of the host to fetch errata for.
+    required: true
+    type: str
+  content_view:
+    description:
+      - Calculate Applicable Errata based on a particular Content View.
+      - Required together with I(lifecycle_environment).
+    required: false
+    type: str
+  lifecycle_environment:
+    description:
+      - Calculate Applicable Errata based on a particular Lifecycle Environment.
+      - Required together with I(content_view).
+    required: false
+    type: str
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.infomodulewithoutname
+'''
+
+EXAMPLES = '''
+- name: "Find errata for host"
```

Comment:
I am supposing you will add an example using environment and content view? Would it make sense to keep this as simple as returning errata for a host and having a separate info module for applicable errata host info? That could keep things separate and simpler and make the playbook a user writes a bit clearer to understand, e.g.

```
host_errata_info
host_applicable_errata_info
```

Just thinking out loud.

Reply:
I can add examples, sure.

This essentially implements the API in https://theforeman.org/plugins/katello/3.18/api/apidoc/v2/host_errata/index.html

So with just hostname, you get "installable" errata (I think? the one you see in the UI when you click the content host) and with CV/LCE you get applicable?

I don't think we need two modules for that.

Reply:
> So with just hostname, you get "installable" errata (I think? the one you see in the UI when you click the content host) and with CV/LCE you get applicable?
> 
> I don't think we need two modules for that.

That is my understanding as well. And if I know anything about Katello, installable and applicable are easy to get oneself confused on. So my suggestion was how we could help make that clear for users when they have some inconsistency with their expectations and are staring at their code confused. Thinking back on it, I often wish we had made that distinction more clear in the API between installable and applicable.

Let's phone a friend.... @jlsherrill  from your perspective how do you think we should lay out the design of returning host errata information?

Reply:
Resolving this, thanks @jlsherrill for weighing in. Examples can be updated now